services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/IntegrationTest.kt

```diff
@@ -26,7 +28,8 @@ import org.testcontainers.utility.MountableFile
 @SpringBootTest
 @ActiveProfiles("test")
 @WithMockUser(USERNAME)
-abstract class DatabaseTest {
+@ExtendWith(MockFileClientExtension::class)
+abstract class IntegrationTest {
```

Comment:
Has it been considered that we would use Azurite as a Test Container like we do for the database?

Reply:
We use an Azurite container for `BlobClientITest` to test the client code against a close approximation of the real Blob Storage.

I think the mock implementation is enough for other tests. The difference is that Blob Storage is a lot less intricate than an SQL server, so it's not hard to mock it, while it would be impossible to mock an SQL server.

It might be OK to use Azurite everywhere. It shouldn't slow the tests down when running all tests, but it might slow down running single tests or test classes, though it might not even be noticeable. If I remember correctly, validating the uploaded files was harder with the Azurite approach, which was one of the reasons we use a mock implementation in most tests.

Reply:
Valid points and no need to change the current implementation. If I remember correctly, there was one issue with Azure Blob Storage that we didn't catch in unit tests. Was it something to do with the scandic letters or other symbols in attachment filenames? Perhaps that sort of errors would be caught using Azurite in a test container. But, no need to anything about it now.